### ISSUE TITLE
Automatic start solution

### ISSUE THREAD

**Issue by dellas60** [Cathwyler/MagicRemoteService#31]
Can you use this https://github.com/JPersson77/LGTVCompanion  for create a script in windows to auto start app in overlay mode? thanks.

**Comment by Cathwyler** [comment]
With the latest release the overlay should now start automatically after a TV reboot and if your TV is starting on the HDMI input but with a 10s-30s delay.
If it's not properly starting can you send a scrennshoot of your TV version ? 
![Sans titre](https://github.com/Cathwyler/MagicRemoteService/assets/4673422/b5d858ba-0ebc-47a4-add8-fbdc6e1da2ce)

**Comment by dellas60** [comment]
<img width="187" alt="Capture d'écran 2024-03-30 214243" src="https://github.com/Cathwyler/MagicRemoteService/assets/29574213/22a04668-398a-47f4-95e5-15ffa68c9419">

**Comment by Cathwyler** [comment]
The overlay should already start and stop automatically depending on the selected video input even after a TV reboot. I fixed several bugs recently that concerned version 4.x of webOS on the service. I now need to confirm if the API used by the service works or not on webOS 4.x

Can you please try these CLI commands lines in a Windows prompt and send me back the returned JSON text?

`ares-novacom -d CATHWYLER-TVSALON -r "luna-send-pub -n 1 'luna://com.webos.service.videooutput/getStatus' '{}'"`

`ares-novacom -d CATHWYLER-TVSALON -r "luna-send-pub -n 1 'luna://com.webos.service.avoutput/video/getStatus' '{}'"`

**Comment by dellas60** [comment]
<img width="593" alt="Capture d'écran 2024-04-01 144410" src="https://github.com/Cathwyler/MagicRemoteService/assets/29574213/a625ddf0-064b-4930-a189-881ae50038eb">
<img width="671" alt="Capture d'écran 2024-04-01 144613jj" src="https://github.com/Cathwyler/MagicRemoteService/assets/29574213/94a3e0ac-f870-4d3b-8b4e-2d9db71af665">

**Comment by Cathwyler** [comment]
Maybe theye are under palm group. Can you try again with palm instead of webos?

`ares-novacom -d TV2 -r "luna-send-pub -n 1 'luna://com.palm.service.videooutput/getStatus' '{}'"`

`ares-novacom -d TV2 -r "luna-send-pub -n 1 'luna://com.palm.service.avoutput/video/getStatus' '{}'"`

**Comment by Cathwyler** [comment]
Can you take a look at the latest release ?
I changed the way the service get the foreground app. He is now using ssap. You need to accept the external device the first time you launch the app.
Can you test if the app is closing and starting when you switch to another app and come back to hdmi?
Can you test if the app is strating after reboot if the hdmi is selected ?

**Comment by dellas60** [comment]
I'm going to test it today when I get home than you for your efforts.

**Comment by dellas60** [comment]
> Can you take a look at the latest release ? I changed the way the service get the foreground app. He is now using ssap. You need to accept the external device the first time you launch the app. Can you test if the app is closing and starting when you switch to another app and come back to hdmi? Can you test if the app is strating after reboot if the hdmi is selected ?

First there is no prompt to accept external device
second the app in not start when close app or reboot

**Comment by Cathwyler** [comment]
To be sure, can you confirm you reinstalled the TV app with the new version before testing it ?

**Comment by dellas60** [comment]
> To be sure, can you confirm you reinstalled the TV app with the new version before testing it ?

Yes, I removed the previous app from the TV and installed the new one from version 1.2.4.8.

**Comment by Cathwyler** [comment]
I fixed one bug that can explain why you don't have any external prompt.

Can you give a new try with the latest.
If it's not working again, can you use [this debug version](https://github.com/Cathwyler/MagicRemoteService/files/14934737/MagicRemoteService_1.2.4.9_Debug.zip) and send me the inspector log.

Btw you need to reinstall TV app each time you update the exe but it seems your are already doing it well.

**Comment by dellas60** [comment]
> I fixed one bug that can explain why you don't have any external prompt.
> 
> Can you give a new try with the latest. If it's not working again, can you use [this debug version](https://github.com/Cathwyler/MagicRemoteService/files/14934737/MagicRemoteService_1.2.4.9_Debug.zip) and send me the inspector log.
> 
> Btw you need to reinstall TV app each time you update the exe but it seems your are already doing it well.

Can you show me how to get the inspector log

**Comment by Cathwyler** [comment]
Click on Inspect button in TV tab in the PC app. A web browser should open. The access to log depend off webOS version.
on 3.x (emulator)
![Sans titre](https://github.com/Cathwyler/MagicRemoteService/assets/4673422/4a4241f9-f1e1-4d83-803b-66ed84ef4d53)

on 6.x (real TV)
![Sans titre](https://github.com/Cathwyler/MagicRemoteService/assets/4673422/4e0a7081-b991-4812-afc6-d87a9214e692)

The 4.x (emulator) inspector is buggy. i have a blank page. I hope it's not the same with real TV.

**Comment by dellas60** [comment]
![photo_2024-04-10_19-49-46](https://github.com/Cathwyler/MagicRemoteService/assets/29574213/43142506-f9fb-49b5-b9bf-e922e10a7f0c)
I GET THIS ERROR IN 1.2.4.9

**Comment by dellas60** [comment]
> i have a blank page

Same i have a blank page and error message failed to close

**Comment by Cathwyler** [comment]
That's so strange i had the same error message as you with the emulator while doing the screenshot and i just get it for the first time. And i'm not able to reproduce it. Do this message appear every time your reinstall the TV app? or once you restart the TV?

I think the webOS inspector is buggy with the 4.x. I will add a better solution to get the log message.

**Comment by dellas60** [comment]
Do this message appear every time your reinstall the TV app? YES or once you restart the TV? YES

Do this message happen only with the release version or the debug one ? release version and debug one

**Comment by dellas60** [comment]
After enter ip adress and click run nothing happen,version 1.2.4.7 WORKS PERFECT WITHOUT AUTORUN, AND RUN MANUALLY THIS IS NOT A PROBLEM FOR ME IF YOU CANNOT FIX THIS PROBLEM THERE IS NO PROBLEM YOU MAKE A GREAT TOOL THANK YOU

**Comment by dellas60** [comment]
CAN YOU WORK WITH AUTO ACTIVATION KEYBOARD WHEN FOCUSING IN TEXT AREA
